core/experiments/assets/js/admin/behaviors/experiments-messages.js

```diff
@@ -168,4 +170,60 @@ export default class ExperimentsDependency {
 
 		return clone.join( glue ) + finalGlue + lastItem;
 	}
+
+	setDialog( experiment, dialog ) {
+		return elementorCommon.dialogsManager.createWidget( 'confirm', {
+			id: 'e-experiments-messages-dialog',
+			headerMessage: dialog.headerMessage,
+			message: dialog.message,
+			position: {
+				my: 'center center',
+				at: 'center center',
+			},
+			strings: {
+				confirm: dialog.confirm,
+				cancel: dialog.cancel,
+			},
+			hide: {
+				onOutsideClick: false,
+				onBackgroundClick: false,
+				onEscKeyPress: false,
+			},
+			onConfirm: () => {
+				dialog.onConfirm();
+			},
+			onCancel: () => {
+				this.setExperimentState( experiment.name, dialog.initialState )
+			},
+		} );
+	}
```

Comment:
maybe it's redundant, you can move it back to the original function

Reply:
DoneMoved in to `showDialog()` as suggested